src/ophys_etl/modules/roi_cell_classifier/compute_classifier_artifacts.py

```diff
@@ -0,0 +1,275 @@
+        """
+        desired_shape = (self.args['cutout_size'], self.args['cutout_size'])
+
+        ophys_roi = extract_roi_to_ophys_roi(extract_roi)
+        pixel_array = ophys_roi.global_pixel_array.transpose()
+
+        # Compute center of cutout from ROI bounding box.
+        if extract_roi['height'] % 2 == 0:
+            center_row = extract_roi['y'] + extract_roi['height'] // 2
+        else:
+            center_row = extract_roi['y'] + extract_roi['height'] // 2 + 1
+        if extract_roi['width'] % 2 == 0:
+            center_col = extract_roi['x'] + extract_roi['width'] // 2
+        else:
+            center_col = extract_roi['x'] + extract_roi['width'] // 2 + 1
+
+        # Get size of cutout.
+        r0 = max(0, center_row - desired_shape[0] // 2)
+        r1 = min(max_img.shape[0], center_row + desired_shape[0] // 2)
+        c0 = max(0, center_col - desired_shape[1] // 2)
+        c1 = min(max_img.shape[1], center_col + desired_shape[1] // 2)
+
+        max_thumbnail = max_img[r0:r1, c0:c1]
+        avg_thumbnail = avg_img[r0:r1, c0:c1]
+        corr_thumbnail = corr_img[r0:r1, c0:c1]
+
+        mask = np.zeros(max_img.shape, dtype=np.uint8)
+
+        # Set value of mask image for cutout.
+        mask[pixel_array[0], pixel_array[1]] = 255
+
+        mask_thumbnail = mask[r0:r1, c0:c1]
+        assert mask_thumbnail.sum() > 0
+
+        # If cutout is on the edge, pad with zeros.
+        need_to_pad = False
+        before_row = 0
+        after_row = 0
+        if r1 - r0 < desired_shape[0]:
+            need_to_pad = True
+            deficit = desired_shape[0] - r1 + r0
+            if r0 == 0:
+                before_row = deficit
+            else:
+                after_row = deficit
+
+        before_col = 0
+        after_col = 0
+        if c1 - c0 < desired_shape[1]:
+            need_to_pad = True
+            deficit = desired_shape[1] - c1 + c0
+            if c0 == 0:
+                before_col = deficit
+            else:
+                after_col = deficit
+
+        # Pad the cutouts if needed.
+        if need_to_pad:
+            padding = ((before_row, after_row), (before_col, after_col))
+            max_thumbnail = np.pad(max_thumbnail,
+                                   pad_width=padding, mode="constant",
+                                   constant_values=0)
+            avg_thumbnail = np.pad(avg_thumbnail,
+                                   pad_width=padding, mode="constant",
+                                   constant_values=0)
+            corr_thumbnail = np.pad(corr_thumbnail,
+                                    pad_width=padding, mode="constant",
+                                    constant_values=0)
+            mask_thumbnail = np.pad(mask_thumbnail,
+                                    pad_width=padding, mode="constant",
+                                    constant_values=0)
+
+        roi_id = ophys_roi.roi_id
+        if mask_thumbnail.sum() <= 0:
```

Comment:
There might be cases where this check will fail, an ROI that is totally dark. I am not sure.